src/jni_core/src/jvm_runtime.rs

```diff
@@ -0,0 +1,291 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use core::option::Option::Some;
+use core::result::Result::{Err, Ok};
+use std::ffi::c_void;
+use std::fs;
+use std::path::Path;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::LazyLock;
+
+use jni::strings::JNIString;
+use jni::{InitArgsBuilder, JNIVersion, JavaVM, NativeMethod};
+use risingwave_common::util::resource_util::memory::total_memory_available_bytes;
+
+use crate::run_this_func_to_get_valid_ptr_from_java_binding;
+
+pub static JVM: LazyLock<Option<JavaVM>> = LazyLock::new(|| {
+    let libs_path = if let Ok(libs_path) = std::env::var("CONNECTOR_LIBS_PATH") {
+        libs_path
+    } else if std::env::var("ENABLE_BUILD_RW_CONNECTOR").is_ok() {
+        // If ENABLE_BUILD_RW_CONNECTOR is set which means it is started from risedev with a
+        // connector.
+        ".risingwave/bin/connector-node/libs/".to_string()
+    } else {
+        return None;
+    };
+
+    let dir = Path::new(&libs_path);
+
+    if !dir.is_dir() {
+        panic!("{} is not a directory", libs_path);
+    }
+
+    let mut class_vec = vec![];
+
+    if let Ok(entries) = fs::read_dir(dir) {
+        for entry in entries.flatten() {
+            if let Some(name) = entry.path().file_name() {
+                class_vec.push(libs_path.to_owned() + name.to_str().to_owned().unwrap());
+            }
+        }
+    } else {
+        panic!("failed to read directory {}", libs_path);
+    }
+
+    let jvm_heap_size = if let Ok(heap_size) = std::env::var("JVM_HEAP_SIZE") {
+        heap_size
+    } else {
+        // Use 10% of total memory by default
+        format!("{}", total_memory_available_bytes() / 10)
+    };
```

Comment:
Another environment variable is `JVM_HEAP_SIZE ` which is used to control the JVM  heap size. By default, I set it to one-tenth of the total memory.